tests/block_tools.py

```diff
@@ -2007,25 +2007,37 @@ def create_test_unfinished_block(
     )
 
 
+create_block_tools_async_count = 0
+create_block_tools_count = 0
+
+
+# xxx do the callers need different instances?
 async def create_block_tools_async(
     constants: ConsensusConstants = test_constants,
     root_path: Optional[Path] = None,
     const_dict=None,
     keychain: Optional[Keychain] = None,
 ) -> BlockTools:
+    global create_block_tools_async_count
+    create_block_tools_async_count += 1
+    print(f"  create_block_tools_async called {create_block_tools_async_count} times")
```

Comment:
We can remove these two print stmts before this patch lands

Reply:
I take it you intend to remove those global counters too, right?

Reply:
I'll remove these when they are not being run multiple times